stl/inc/filesystem

```diff
@@ -1377,6 +1377,36 @@ namespace filesystem {
 #endif // ^^^ !_HAS_CXX20 ^^^
 
         _NODISCARD_FRIEND path operator/(const path& _Left, const path& _Right) { // append a pair of paths together
+            const auto _Right_size  = _Right._Text.size();
+            const auto _Right_first = _Right._Text.data();
+            const auto _Right_last  = _Right_first + _Right_size;
+
+            // Handle the most common case
+            if (!_Has_drive_letter_prefix(_Right_first, _Right_last) && !_Is_slash(*_Right_first)) {
+                // !has_root_name(_Right) && !has_root_directory(_Right)
+                const auto _Left_size  = _Left._Text.size();
+                const auto _Left_first = _Left._Text.data();
+                const auto _Left_last  = _Left._Text.data() + _Left_size;
+
+                // Appending a slash to "X:" would've made it an absolute path
+                const bool _Left_is_just_drive = _Left_size == 2 && _Is_drive_prefix(_Left_first);
+                const bool _Is_slash_needed    = _Left_size != 0 && !_Left_is_just_drive && !_Is_slash(_Left_last[-1]);
+
+                const auto _Total_size = _Left_size + _Is_slash_needed + _Right_size;
+
+                path _Tmp;

```

Comment:
Returning two different local variables breaks the [Named Return Value Optimization](https://en.cppreference.com/w/cpp/language/copy_elision#Non-mandatory_copy.2Fmove_.28since_C.2B.2B11.29_elision). Can we move this definition of `_Tmp` above the `if`, and assign `_Left` to it on 1410 instead of copy constructing?

Reply:
Are you sure that:

1. Even non-overlapping lifetimes of multiple return variables would inhibit NVRO?
2. Even if NVRO would not work, move return would impact more than default-constructing?

Reply:
No, I am not. Since this is a perf change, someone needs to at least examine the codegen both ways.

Reply:
I looked at the codegen (although disclaimer, I am mostly assembly-illiterate). I *think* the NRVO is still active despite the multiple named `_Tmp` variables. However, if I try to unify them, I definitely see `basic_string`'s copy assignment operator being emitted (for `_Tmp = _Left;`), and that appears to be way more code than the copy construction.

I think we should keep this as-is.